lld/ELF/SyntheticSections.cpp

```diff
@@ -668,6 +668,8 @@ void GotSection::addConstant(const Relocation &r) { relocations.push_back(r); }
 void GotSection::addEntry(const Symbol &sym) {
   assert(sym.auxIdx == ctx.symAux.size() - 1);
   ctx.symAux.back().gotIdx = numEntries++;
+  if (sym.hasFlag(NEEDS_GOT_AUTH))
```

Comment:
should be moved to `addGotAuthEntry`

Reply:
Fixed in 77361c75f4e6e8b99121f8a059a48593eb6a754d